include/drivers/dma.h

```diff
@@ -327,6 +364,78 @@ static inline int z_impl_dma_stop(const struct device *dev, uint32_t channel)
 	return api->stop(dev, channel);
 }
 
+/**
+ * @brief request DMA channel.
+ *
+ * request DMA channel resources
+ * return -EINVAL if there is no valid channel available.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param filter_param filter function parameter
+ *
+ * @retval dma channel if successful.
+ * @retval Negative errno code if failure.
+ */
+__syscall int dma_request_channel(const struct device *dev,
+				  void *filter_param);
+
+static inline int z_impl_dma_request_channel(const struct device *dev,
+					     void *filter_param)
+{
+	int i = 0;
+	int channel = -EINVAL;
+	const struct dma_driver_api *api =
+		(const struct dma_driver_api *)dev->api;
+	/*dma_context shall be the first one in dev data*/
```

Comment:
This is a pretty strong assumption, this should not be an hidden comment.

Reply:
what do you recommand? this is to isolate the low layer driver with dma framework. I did not find a better way to get the dma_context.

Reply:
Yes, at least that please.

Reply:
Thanks updated.